[V3/Readme] Update V3 Readme

Redjumpman opened the pull request:
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Reference Issue: #1702
Preview: [link](https://github.com/Redjumpman/Red-DiscordBot/tree/V3/readme)

Updated the readme to be more aligned with the polish of V2's readme. This is likely to be an on-going enhancement and will be subject to several different opinions. While many things can be added or changed in the future, I think that it is paramount to include the following:

- What Red V3 offers
- Links for installing on all supported platforms
- License information
- How people can join the community
- How/Why you should upgrade from V2

I think the last part is critical for a higher adoption rate. I have hit on some points for that (how to transfer your data (link in the docs), but an additional blurb in the docs about what V3 does better would be a great addition later down the line.

The cogs that I included as the Third Party are cogs from random people (and shamelessly myself) that are already **V3** compatible. Feel free to change that or give suggestions.

As for the red portal links, I am currently using the provisional `http://dev.v3.cogs.red/` link. This can be updated as Orels gets closer to a V3 full release.

Redjumpman committed:
Update and rename README.rst to README.MD

Redjumpman committed:
Update and rename README.MD to README.rst

Changed the file type back to rst from MD

Redjumpman commented:
Changed the file type back to rst.

palmtree5 commented:
I'm thinking that for now it might be better to just point the link to dev.v3.cogs.red to #1398 instead because dev.v3.cogs.red is only displaying V2 cogs atm, and I feel linking to a site that is displaying cogs written for V2 would just serve to confuse users. The link can of course be updated when that site is displaying V3 cogs though

Redjumpman committed:
Update README.rst

Changed image size.

palmtree5 committed:
Merge branch 'V3/develop' into V3/readme

Redjumpman committed:
Update README.rst

Changed the cogs.red link to point at issue 1398 until the portal displays V3 cogs.

Redjumpman commented:
Changed the portal links to point to #1398 .

palmtree5 committed:
Merge branch 'V3/develop' into V3/readme

Redjumpman committed:
Update README.rst

Changed image host to imgur

palmtree5 committed:
Merge branch 'V3/develop' into V3/readme

palmtree5 committed:
Merge branch 'V3/develop' into V3/readme
